packages/core/src/modules/credentials/CredentialsModule.ts

```diff
@@ -354,6 +355,14 @@ export class CredentialsModule<
     const service = this.getService(credentialRecord.protocolVersion)
     await service.updateState(credentialRecord, CredentialState.Declined)
 
+    //  if credential was offered by established connection, attempt to send problem report upon credential decline
```

Comment:
There have been discussions in the past on whether sending a problem report is desired as it leaks some info you may not want to share with the other party. I think we should therefore make it something configurable by e.g. adding a `sendProblemReport` option:

```ts
await agent.credentials.declineOffer({
  credentialRecordId: 'xxx',
  sendProblemReport: true,
  problemReportMessage: 'something' # optional message
```